upf/upf_api.c

```diff
@@ -1061,8 +1061,44 @@ vl_api_upf_tdf_ul_table_add_t_handler (vl_api_upf_tdf_ul_table_add_t * mp)
   REPLY_MACRO (VL_API_UPF_TDF_UL_TABLE_ADD_REPLY);
 }
 
-#include <upf/upf.api.c>
+/* API message handler */
+static void
+vl_api_upf_nat_pool_add_t_handler (vl_api_upf_nat_pool_add_t * mp)
+{
+  int rv = 0;
+  upf_main_t *sm = &upf_main;
+  vl_api_upf_nat_pool_add_reply_t *rmp = NULL;
+
+  u8 *name_vec = 0;
+  u8 *nwi_vec = 0;
+  u32 name_vec_len = 0;
+  ip4_address_t start, end;
+
+  mp->min_port = ntohs (mp->min_port);
+  mp->max_port = ntohs (mp->max_port);
+  mp->block_size = ntohl (mp->block_size);
 
+  ASSERT (mp->nwi_len <= 64);
```

Comment:
This may cause an assertion failure due to bad API call parameters, which is not a correct behavior. The proper behavior here would be returning an error from the API call. Verifying such a situation in the E2E test would also be nice.